engine/language-server/src/main/scala/org/enso/languageserver/boot/ProfilingConfig.scala

```diff
@@ -23,19 +23,57 @@ case class ProfilingConfig(
     * @return the path to the runtime events log file
     */
   def profilingEventsLogPath: Option[Path] =
-    profilingPath.map { path =>
-      val profilingDirectory     = path.getParent
-      val profilingFileName      = path.getFileName.toString
-      val profilingFileExtension = FilenameUtils.getExtension(profilingFileName)
-      val eventsLogFileName =
-        profilingFileName.stripSuffix(
-          profilingFileExtension
-        ) + ProfilingConfig.EventsLogExtension
-
-      profilingDirectory.resolve(eventsLogFileName)
-    }
+    profilingPath.map(
+      ProfilingConfig.modifyPath(
+        _,
+        ProfilingConfig.EventsLogSuffix,
+        ProfilingConfig.EventsLogExtension
+      )
+    )
+
+  /** Creates a path to the messages log. */
+  def messagesPath: Option[Path] =
+    profilingPath.map(
+      ProfilingConfig.modifyPath(
+        _,
+        ProfilingConfig.MessagesSuffix,
+        ProfilingConfig.MessagesExtension
+      )
+    )
 }
+
 object ProfilingConfig {
 
+  private val EventsLogSuffix    = ""
   private val EventsLogExtension = "log"
+
+  private val MessagesSuffix    = "-messages"
+  private val MessagesExtension = "txt"
```

Comment:
I'd avoid this `.txt` file and instead I'd write the messages directly into the `.log` file together with other `EventsLog` events next to `.npss`.

When opening the `.npss` file VisualVM will pick up also the `.log` file and display all the events in its **timeline**. One can **zoom** the timeline, find a particular event interval and then see only **samples from that period**. That might be pretty useful.

Reply:
Do you have a screenshot or sample `.npss` and `.log` to show how the integration works?

Reply:
Here is an example [language-server-prifiling.zip](https://github.com/user-attachments/files/15510811/language-server-prifiling.zip)

Reply:
I can zoom, I can see `AttachVisualization` message and I can inspect the profiling to tell me that 30ms takes to parse the message by Jackson:

![timeline in VisualVM](https://github.com/enso-org/enso/assets/26887752/8f0b03fb-523c-4f58-a023-87a1247586f3)

I consider this useful for the use-cases I can imagine.